src/torchmetrics/regression/pearson.py

```diff
@@ -151,6 +151,7 @@ def update(self, preds: Tensor, target: Tensor) -> None:
             self.corr_xy,
             self.n_total,
             self.num_outputs,
+            self.weights,
```

Comment:
```suggestion
            weights=weights,
```
this is probably why CI is failing.
You need to update the `update` signature here:
https://github.com/Lightning-AI/torchmetrics/blob/4e0ef77fd722f1f332b557b34bfec05b24e6b31a/src/torchmetrics/regression/pearson.py#L142-L143
such that is also takes in the `weights` argument.

Reply:
I included the suggested changes and have been debugging the tests for the functional implementation `pearson_corrcoef` and am continuing to see `TypeError: pearson_corrcoef() got an unexpected keyword argument 'weights'` even though I have defined

`def pearson_corrcoef(preds: Tensor, target: Tensor, weights: Optional[Tensor] = None) -> Tensor:`

https://github.com/matsumotosan/metrics/blob/4e0ef77fd722f1f332b557b34bfec05b24e6b31a/src/torchmetrics/functional/regression/pearson.py#L119

When I check the imported function, it does not list `weights` as an argument. I've posted the full log for a single case of `test_pearson_corrcoef_functional`.

```
pytest test_pearson.py::TestPearsonCorrCoef::test_pearson_corrcoef_functional[metric_args1-preds0-target0]
[W socket.cpp:601] [c10d] The client socket has failed to connect to [localhost]:8088 (errno: 99 - Cannot assign requested address).
======================================================== test session starts =========================================================
platform linux -- Python 3.9.16, pytest-7.2.2, pluggy-1.0.0
rootdir: /workspaces/metrics, configfile: pyproject.toml
plugins: cov-4.0.0, doctestplus-0.12.1, timeout-2.1.0, rerunfailures-11.1.1
collected 1 item

test_pearson.py F                                                                                                              [100%]

============================================================== FAILURES ==============================================================
_________________________ TestPearsonCorrCoef.test_pearson_corrcoef_functional[metric_args1-preds0-target0] __________________________

self = <unittests.regression.test_pearson.TestPearsonCorrCoef object at 0x7ff767935520>
preds = tensor([[0.8823, 0.9150, 0.3829, 0.9593, 0.3904, 0.6009, 0.2566, 0.7936, 0.9408,
         0.1332, 0.9346, 0.5936, 0.86...716, 0.3336, 0.5782, 0.0600, 0.2846, 0.2007, 0.5014, 0.3139, 0.4654,
         0.1612, 0.1568, 0.2083, 0.3289, 0.1054]])
target = tensor([[0.9192, 0.4008, 0.9302, 0.6558, 0.0766, 0.8460, 0.3624, 0.3083, 0.0850,
         0.0029, 0.6431, 0.3908, 0.69...994, 0.2122, 0.4089, 0.1481, 0.1733, 0.6659, 0.3514, 0.8087, 0.3396,
         0.1332, 0.4118, 0.2576, 0.3470, 0.0240]])
metric_args = {'weights': tensor([[0.7322, 0.6075, 0.3988, 0.7365, 0.6829, 0.0499, 0.2046, 0.5168, 0.6897,
         0.7433, 0.2024, ...92, 0.6699, 0.9007, 0.4766, 0.0543, 0.3549, 0.8514, 0.8044, 0.6548,
         0.3533, 0.0333, 0.2313, 0.7607, 0.2983]])}

    def test_pearson_corrcoef_functional(self, preds, target, metric_args):
        """Test functional implementation of metric."""
>       self.run_functional_metric_test(
            preds=preds,
            target=target,
            metric_functional=pearson_corrcoef,
            reference_metric=partial(_ref_metric, metric_args=metric_args),
            **metric_args
        )

test_pearson.py:119:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../helpers/testers.py:353: in run_functional_metric_test
    _functional_test(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

preds = tensor([[0.8823, 0.9150, 0.3829, 0.9593, 0.3904, 0.6009, 0.2566, 0.7936, 0.9408,
         0.1332, 0.9346, 0.5936, 0.86...716, 0.3336, 0.5782, 0.0600, 0.2846, 0.2007, 0.5014, 0.3139, 0.4654,
         0.1612, 0.1568, 0.2083, 0.3289, 0.1054]])
target = tensor([[0.9192, 0.4008, 0.9302, 0.6558, 0.0766, 0.8460, 0.3624, 0.3083, 0.0850,
         0.0029, 0.6431, 0.3908, 0.69...994, 0.2122, 0.4089, 0.1481, 0.1733, 0.6659, 0.3514, 0.8087, 0.3396,
         0.1332, 0.4118, 0.2576, 0.3470, 0.0240]])
metric_functional = <function pearson_corrcoef at 0x7ff76a519040>
reference_metric = functools.partial(<function _ref_metric at 0x7ff7679014c0>, metric_args={'weights': tensor([[0.7322, 0.6075, 0.3988, 0...2, 0.6699, 0.9007, 0.4766, 0.0543, 0.3549, 0.8514, 0.8044, 0.6548,
         0.3533, 0.0333, 0.2313, 0.7607, 0.2983]])})
metric_args = {}, atol = 0.001, device = 'cpu', fragment_kwargs = False
kwargs_update = {'weights': tensor([[0.7322, 0.6075, 0.3988, 0.7365, 0.6829, 0.0499, 0.2046, 0.5168, 0.6897,
         0.7433, 0.2024, ...92, 0.6699, 0.9007, 0.4766, 0.0543, 0.3549, 0.8514, 0.8044, 0.6548,
         0.3533, 0.0333, 0.2313, 0.7607, 0.2983]])}
p_size = 4, t_size = 4, num_batches = 4, metric = functools.partial(<function pearson_corrcoef at 0x7ff76a519040>)
extra_kwargs = {'weights': tensor([0.7322, 0.6075, 0.3988, 0.7365, 0.6829, 0.0499, 0.2046, 0.5168, 0.6897,
        0.7433, 0.2024, 0....8651, 0.1518, 0.7000, 0.5596, 0.2499, 0.7479, 0.0080, 0.9042, 0.5403,
        0.5483, 0.7312, 0.5415, 0.7143, 0.8181])}

    def _functional_test(
        preds: Union[Tensor, list],
        target: Union[Tensor, list],
        metric_functional: Callable,
        reference_metric: Callable,
        metric_args: dict = None,
        atol: float = 1e-8,
        device: str = "cpu",
        fragment_kwargs: bool = False,
        **kwargs_update: Any,
    ):
        """Comparison between functional metric and reference metric.

        Args:
            preds: torch tensor with predictions
            target: torch tensor with targets
            metric_functional: metric functional that should be tested
            reference_metric: callable function that is used for comparison
            metric_args: dict with additional arguments used for class initialization
            atol: absolute tolerance used for comparison of results
            device: determine which device to run on, either 'cuda' or 'cpu'
            fragment_kwargs: whether tensors in kwargs should be divided as `preds` and `target` among processes
            kwargs_update: Additional keyword arguments that will be passed with preds and
                target when running update on the metric.
        """
        p_size = preds.shape[0] if isinstance(preds, Tensor) else len(preds)
        t_size = target.shape[0] if isinstance(target, Tensor) else len(target)
        assert p_size == t_size, f"different sizes {p_size} and {t_size}"
        num_batches = p_size
        metric_args = metric_args or {}
        metric = partial(metric_functional, **metric_args)

        # move to device
        if isinstance(preds, Tensor):
            preds = preds.to(device)
        if isinstance(target, Tensor):
            target = target.to(device)
        kwargs_update = {k: v.to(device) if isinstance(v, Tensor) else v for k, v in kwargs_update.items()}

        for i in range(num_batches // 2):
            extra_kwargs = {k: v[i] if isinstance(v, Tensor) else v for k, v in kwargs_update.items()}
>           tm_result = metric(preds[i], target[i], **extra_kwargs)
E           TypeError: pearson_corrcoef() got an unexpected keyword argument 'weights'

../helpers/testers.py:271: TypeError
====================================================== short test summary info =======================================================
FAILED test_pearson.py::TestPearsonCorrCoef::test_pearson_corrcoef_functional[metric_args1-preds0-target0] - TypeError: pearson_corrcoef() got an unexpected keyword argument 'weights'
```

Reply:
Alright, I try taking a look at it because it sounds weird to me that you are getting such an error

Reply:
should be fixed now